### ISSUE TITLE
silx.gui.plot breaks with pyfftw

### ISSUE THREAD

**Issue by alexbjorling** [silx-kit/silx#2375]
I just cloned the `silx` master branch to get access to a new feature.

There's some problem with `pyfftw` as shown below. If I uninstall `pyfftw` there's no problem. If I revert to `silx==0.9.0` with `pyfftw` installed, it works too.

```
In [1]: import silx.gui.plot
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-8a6d86e68a2f> in <module>()
----> 1 import silx.gui.plot

/home/alex/.local/lib/python2.7/site-packages/silx/gui/plot/__init__.py in <module>()
     61 
     62 
---> 63 from .PlotWidget import PlotWidget  # noqa
     64 from .PlotWindow import PlotWindow, Plot1D, Plot2D  # noqa
     65 from .items.axis import TickMode

/home/alex/.local/lib/python2.7/site-packages/silx/gui/plot/PlotWidget.py in <module>()
     50 from .backends.BackendMatplotlib import BackendMatplotlibQt
     51 
---> 52 from ..colors import Colormap
     53 from .. import colors
     54 from . import PlotInteraction

/home/alex/.local/lib/python2.7/site-packages/silx/gui/colors.py in <module>()
     37 from silx.gui import qt
     38 from silx import config
---> 39 from silx.math.combo import min_max
     40 from silx.math.colormap import cmap as _cmap
     41 from silx.utils.exceptions import NotEditableError

/home/alex/.local/lib/python2.7/site-packages/silx/math/__init__.py in <module>()
     38 from .histogram import HistogramndLut  # noqa
     39 from .medianfilter import medfilt, medfilt1d, medfilt2d
---> 40 from .fft import fft

/home/alex/.local/lib/python2.7/site-packages/silx/math/fft/__init__.py in <module>()
      6 __date__ = "12/12/2018"
      7 
----> 8 from .fft import FFT

/home/alex/.local/lib/python2.7/site-packages/silx/math/fft/fft.py in <module>()
     24 #
     25 # ###########################################################################*/
---> 26 from .fftw import FFTW
     27 from .clfft import CLFFT
     28 from .npfft import NPFFT

/home/alex/.local/lib/python2.7/site-packages/silx/math/fft/fftw.py in <module>()
     28 from .basefft import BaseFFT, check_version
     29 try:
---> 30     import pyfftw
     31     __have_fftw__ = True
     32 except ImportError:

/home/alex/.local/lib/python2.7/site-packages/pyfftw/__init__.pyc in <module>()
     41 from . import config
     42 from . import builders
---> 43 from . import interfaces
     44 
     45 

/home/alex/.local/lib/python2.7/site-packages/pyfftw/interfaces/__init__.py in <module>()
    240 fft_wrap = None
    241 try:
--> 242     from dask.array.fft import fft_wrap
    243 except ImportError:
    244     pass

/home/alex/.local/lib/python2.7/site-packages/dask/array/__init__.py in <module>()
      7                        from_delayed, asarray, asanyarray, PerformanceWarning,
      8                        broadcast_arrays, broadcast_to, from_zarr, to_zarr)
----> 9     from .routines import (take, choose, argwhere, where, coarsen, insert,
     10                            ravel, roll, unique, squeeze, ptp, diff, ediff1d,
     11                            gradient, bincount, digitize, histogram, cov, array,

/home/alex/.local/lib/python2.7/site-packages/dask/array/routines.py in <module>()
    254 
    255 
--> 256 @wraps(np.matmul)
    257 def matmul(a, b):
    258     a = asanyarray(a)

/usr/lib/python2.7/functools.pyc in update_wrapper(wrapper, wrapped, assigned, updated)
     31     """
     32     for attr in assigned:
---> 33         setattr(wrapper, attr, getattr(wrapped, attr))
     34     for attr in updated:
     35         getattr(wrapper, attr).update(getattr(wrapped, attr, {}))

AttributeError: 'numpy.ufunc' object has no attribute '__module__'
```

**Comment by pierrepaleo** [comment]
Hi @alexbjorling, thanks for the report.
However I am not able to reproduce this bug. My guess is that it looks more like an issue with fftw/dask/numpy/python2.7 which would have been solved with recent versions.
Here is my python 2.7.13 environment:

<details><summary>Installed packages (fresh 2.7.13 virtualenv)</summary>
<pre>
backports.functools-lru-cache==1.5
backports.shutil-get-terminal-size==1.0.0
cycler==0.10.0
Cython==0.29.3
dask==1.1.0
decorator==4.3.2
enum34==1.1.6
fabio==0.8.0
futures==3.2.0
h5py==2.9.0
ipython==5.8.0
ipython-genutils==0.2.0
kiwisolver==1.0.1
matplotlib==2.2.3
numpy==1.16.0
pathlib2==2.3.3
pexpect==4.6.0
pickleshare==0.7.5
prompt-toolkit==1.0.15
ptyprocess==0.6.0
pyFFTW==0.11.1
Pygments==2.3.1
pyparsing==2.3.1
PySide2==5.12.0
python-dateutil==2.7.5
pytz==2018.9
scandir==1.9.0
scipy==1.2.0
shiboken2==5.12.0
silx==0.10.0a0
simplegeneric==0.8.1
six==1.12.0
subprocess32==3.5.3
toolz==0.9.0
traitlets==4.3.2
wcwidth==0.1.7
</pre>
</details>
  
Can you indicate the installed versions of `fftw`, `dask` and `numpy` on your side (`python2 -m pip freeze`) ?

**Comment by alexbjorling** [comment]
Here's the environment. I assumed pip-installing silx would resolve its dependencies.
```
apptools==4.4.0
asn1crypto==0.24.0
attrs==17.4.0
autobahn==17.10.1
Automat==0.6.0
backports.functools-lru-cache==1.5
backports.shutil-get-terminal-size==1.0.0
beautifulsoup4==4.6.0
cbor==1.0.0
certifi==2018.1.18
chardet==3.0.4
click==6.7
cloudpickle==0.6.1
colorama==0.3.7
configobj==5.0.6
constantly==15.1.0
cryptography==2.1.4
cycler==0.10.0
Cython==0.26.1
dask==1.0.0
decorator==4.3.0
enum34==1.1.6
envisage==4.7.0
fabio==0.8.0
futures==3.2.0
h5py==2.9.0
hdf-compass==0.6.0
html5lib==0.999999999
hyperlink==17.3.1
idna==2.6
incremental==16.10.1
ipaddress==1.0.17
ipdb==0.11
ipython==5.8.0
ipython-genutils==0.2.0
keyring==10.6.0
keyrings.alt==3.0
kiwisolver==1.0.1
lxml==4.2.1
lz4==0.10.1
matplotlib==2.2.3
mayavi==4.5.0
mpi4py==3.0.0
NanoMAX-Analysis-Utilities==0.1a0
networkx==2.2
numpy==1.16.0
olefile==0.45.1
PAM==0.4.2
pathlib2==2.3.3
pexpect==4.6.0
pickleshare==0.7.5
Pillow==5.4.1
prompt-toolkit==1.0.15
ptyprocess==0.6.0
py-ubjson==0.8.5
pyasn1==0.4.2
pyasn1-modules==0.2.1
pycairo==1.16.2
pycparser==2.19
pycrypto==2.6.1
pyface==6.0.0
PyFFTW3==0.2.2
Pygments==2.3.1
pygobject==3.26.1
PyNaCl==1.1.2
pyOpenSSL==17.5.0
pyparsing==2.3.1
pyserial==3.4
python-dateutil==2.7.5
Python-Ptychography-toolbox==0.3.0
python-snappy==0.5
PyTrie==0.2
pytz==2018.9
PyWavelets==1.0.1
pyxdg==0.25
pyzmq==17.1.2
qrcode==5.3
requests==2.18.4
scandir==1.9.0
scikit-image==0.14.1
scipy==1.2.0
scour==0.36
SecretStorage==2.3.1
service-identity==16.0.0
silx==0.10.0a0
simplegeneric==0.8.1
six==1.12.0
subprocess32==3.5.3
toolz==0.9.0
traitlets==4.3.2
traits==4.6.0
traitsui==6.0.0
trollius==2.0.1
Twisted==17.9.0
txaio==2.8.1
u-msgpack-python==2.1
urllib3==1.22
wcwidth==0.1.7
webencodings==0.5
wsaccel==0.6.2
wxPython==3.0.2.0
wxPython-common==3.0.2.0
zope.interface==4.3.2
```

**Comment by t20100** [comment]
I would remove `from .fft import fft` from `silx/math:__init__.py` (https://github.com/silx-kit/silx/blob/master/silx/math/__init__.py#L40) as this results in importing all `silx.math.fft` dependencies each time one is using something else from `silx.math` like the plot stuff.

**Comment by vasole** [comment]
@alexbjorling 

Can the issue be considered closed by #2384?

**Comment by alexbjorling** [comment]
Sorry for the delay. Yes, this problem is resolved.
